crates/arrow_sql_gen/src/statement.rs

```diff
@@ -55,6 +65,181 @@ impl CreateTableBuilder {
     }
 }
 
+pub struct InsertBuilder {
+    table_name: String,
+    record_batches: Vec<RecordBatch>,
+}
+
+impl InsertBuilder {
+    #[must_use]
+    pub fn new(table_name: &str, record_batches: Vec<RecordBatch>) -> Self {
+        Self {
+            table_name: table_name.to_string(),
+            record_batches,
+        }
+    }
+
+    #[allow(clippy::too_many_lines)]
+    pub fn construct_insert_stmt(
```

Comment:
FYI, no action needed: This will be a useful fallback to databases we support which don't have a more efficient way of inserting data - but for Postgres specifically, we can leverage the binary protocol. @y-f-u is working on that.

Reply:
there is a copy_into method that create a sink as well as a binary in thing we can utilise.

The implementation will be quite straight forward, just FYI. I mentioned it previous in the ticket, but seems it's removed now. I will create a card and assign it to me.